### PR TITLE
Improve getting datas

### PR DESCRIPTION
**Pull request by cpaillet**
Improve timers.
- get only the last analytics dataset
- get logs form 2 minutes ago.
+1 small Typo

### CODE REVIEW

File: cloudflare_exporter/collector.py [hunk 1]
```diff
@@ -177,9 +177,9 @@ def add(self, zone, serie):
 
 def _get_cloudflare_metrics_from_logs(token, logs_count, logs_sample, logs_range):
     cloudflare = CloudFlare.CloudFlare(debug=False, token=token)
-    # from cf docs: Must be at least 1 minute earlier than now and later than start
-    # Sometime 1 minutes is not enough...
-    end = int(time.time() - 60 * 1)
+    unixtime_lastminute = int(time.time() / 60) * 60
```
Comment (dclaisse) [inline cloudflare_exporter/collector.py]: (x / 60) * 60 == x? What does int casting bring?

File: cloudflare_exporter/collector.py [hunk 2]
```diff
@@ -94,13 +94,10 @@ def collect(self):
...
         if self.logs_fetch:
             self.logs_collect(families)
-
         for zone, raw_data in _get_cloudflare_analytics(self.cloudflare_token):
-            # We're interested in the latest metrics, however
-            # the Cloudflare API doesn't guarantee non-zero values.
-            # Index -2 was chosen empirically and is usually non-zero.
             for pop_data in raw_data:
-                serie = pop_data['timeseries'][-2]
+                # We use the most recent timeserie
```
Comment (dclaisse) [inline cloudflare_exporter/collector.py]: This comment seems to become useless.

File: cloudflare_exporter/collector.py [hunk 3]
```diff
@@ -142,9 +139,12 @@ def collect(self):
...
     cloudflare = CloudFlare.CloudFlare(debug=False, token=token)
     for zone in cloudflare.zones.get():
-        # get the last 30 minutes (the very last 5 minutes are usually not correct)
         try:
-            series = cloudflare.zones.analytics.colos(zone['id'], params={'since': -35, 'until': -5})
+            # from https://api.cloudflare.com/#zone-analytics-dashboard
+            # By default, range for the Last 60 minutes (from -59 to -1): 1 minute resolution
+            # by default-> the API will move the requested time window backward,
+            #              until it finds a region with completely aggregated data.
+            series = cloudflare.zones.analytics.colos(zone['id'], params={'since': -1})
```
Comment (dclaisse) [inline cloudflare_exporter/collector.py:152]: Is that really what you want? I'm reading "since 1 minute ago".

**Review by dclaisse — DISMISSED** [review]

File: cloudflare_exporter/collector.py (hunk 3 above)
Comment (cpaillet) [reply cloudflare_exporter/collector.py:152]: I update the comment

File: cloudflare_exporter/collector.py (hunk 2 above)
Comment (cpaillet) [reply cloudflare_exporter/collector.py]: comment updated

File: cloudflare_exporter/collector.py (hunk 1 above)
Comment (cpaillet) [reply cloudflare_exporter/collector.py]: start at the minute  ( cloudflare always return a full minutes range ( from 0s to 59s )

File: cloudflare_exporter/collector.py (hunk 3 above)
Comment (pierrecdn) [reply cloudflare_exporter/collector.py:152]: OPT:
Can't see the first patchset, but IMO providing Cloudlfare API link is sufficient.
Eventually you can keep "Retrieve last minute analytics, aggregated using 1-minute resolution"

File: cloudflare_exporter/collector.py (hunk 1 above)
Comment (pierrecdn) [reply cloudflare_exporter/collector.py]:
```
from datetime import datetime, timedelta
past_1minute = datetime.now() - timedelta(minutes=1)
unixtime = int(past_1minute.timestamp())
```
makes it more readable or at least more conventional (your intent is not ambiguous).

File: cloudflare_exporter/collector.py [hunk 4]
```diff
@@ -177,9 +178,10 @@ def add(self, zone, serie):
 
 def _get_cloudflare_metrics_from_logs(token, logs_count, logs_sample, logs_range):
     cloudflare = CloudFlare.CloudFlare(debug=False, token=token)
-    # from cf docs: Must be at least 1 minute earlier than now and later than start
-    # Sometime 1 minutes is not enough...
-    end = int(time.time() - 60 * 1)
+    unixtime_lastminute = int(time.time() / 60) * 60
+    # from cf docs: Must be at least 5 minute earlier than now and later than start
+    # https://developers.cloudflare.com/logs/logpull-api/requesting-logs/ ( 5minutes )
+    end = unixtime_lastminute - 60 * 5
```
Comment (pierrecdn) [inline cloudflare_exporter/collector.py]: same here.

**Review by pierrecdn — COMMENTED** [review]
"Improve getting datas": what do you mean? data quality? accuracy? performance or efficiency in retrieving datas?
You may provide more context so that reviewers can clearly see what are you doing generally speaking and why it make to group these 3 commits together.

File: cloudflare_exporter/collector.py (hunk 1 above)
Comment (cpaillet) [reply cloudflare_exporter/collector.py]: yes better

File: cloudflare_exporter/collector.py (hunk 3 above)
Comment (cpaillet) [reply cloudflare_exporter/collector.py:152]: rebase mistake, it's part of the first patchset

File: cloudflare_exporter/collector.py [hunk 5]
```diff
@@ -178,8 +179,10 @@ def add(self, zone, serie):
 def _get_cloudflare_metrics_from_logs(token, logs_count, logs_sample, logs_range):
     cloudflare = CloudFlare.CloudFlare(debug=False, token=token)
     # from cf docs: Must be at least 1 minute earlier than now and later than start
-    # Sometime 1 minutes is not enough...
-    end = int(time.time() - 60 * 1)
+    # https://developers.cloudflare.com/logs/logpull-api/requesting-logs/
+    # Sometime 1 minutes seems not enought
```
Comment (dclaisse) [inline cloudflare_exporter/collector.py]: typo: enough

**Comment by dclaisse** [comment]
Title of PR doesn't look meaningful to me, could you please explain what it does?

**Comment by cpaillet** [comment]
you don't see the PR comment ?

**Review by pierrecdn — APPROVED** [review]
Next time, please open a PR per topic, it's not easy for people to review this: bugfixes, version upgrade, changes mixed together are not related to "improve getting datas" (data BTW).

**Pull request merged**